zookeeper-server/src/test/java/org/apache/zookeeper/test/BackupManagerTest.java

```diff
@@ -0,0 +1,601 @@
+import org.apache.zookeeper.server.ServerCnxnFactory;
+import org.apache.zookeeper.server.SyncRequestProcessor;
+import org.apache.zookeeper.server.ZooKeeperServer;
+import org.apache.zookeeper.server.backup.BackupConfig;
+import org.apache.zookeeper.server.backup.BackupFileInfo;
+import org.apache.zookeeper.server.backup.BackupManager;
+import org.apache.zookeeper.server.backup.BackupPoint;
+import org.apache.zookeeper.server.backup.BackupStatus;
+import org.apache.zookeeper.server.backup.BackupUtil;
+import org.apache.zookeeper.server.backup.BackupUtil.BackupFileType;
+import org.apache.zookeeper.server.backup.storage.impl.LocalBackupStorage;
+import org.apache.zookeeper.server.persistence.FileTxnSnapLog;
+import org.apache.zookeeper.server.persistence.Util;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests BackupManager. This class is intended to be in the package "org.apache.zookeeper.test" to
+ * leverage other ZK testing components such as ClientBase.
+ */
+public class BackupManagerTest extends ZKTestCase implements Watcher {
+
+  private static String HOSTPORT = "127.0.0.1:" + PortAssignment.unique();
+  private static final int CONNECTION_TIMEOUT = 300000;
+  private static final Logger LOG = LoggerFactory.getLogger(BackupManagerTest.class);
+
+  private ZooKeeper connection;
+  private File dataDir;
+  private File backupStatusDir;
+  private File backupTmpDir;
+  private File backupDir;
+  private ZooKeeperServer zks;
+  private ServerCnxnFactory serverCnxnFactory;
+  private LocalBackupStorage backupStorage;
+  private BackupManager backupManager;
+  private BackupStatus backupStatus;
+  private FileTxnSnapLog snapLog;
+  private int createdNodeCount = 0;
+  private BackupConfig backupConfig;
+
+  @Before
+  public void setup() throws Exception {
+    // Create a dummy config
+    backupConfig = new BackupConfig.Builder().
+        setEnabled(true).
+        setStatusDir(new File("/backup_manager_test/backup/status")).
+        setTmpDir(new File("/backup_manager_test/tmp/backup")).
+        setMountPath("/backup_manager_test").
+        setNamespace("/test_namespace").
+        build().get();
+
+    setupZk();
+    connection = createConnection();
+  }
+
+  @After
+  public void teardown() throws Exception {
+    closeConnection(connection);
+    connection = null;
+    cleanupZk();
+  }
+
+  @Test
+  public void testEmptyDirectory() throws Exception {
+    File dataDir = ClientBase.createTmpDir();
+    File backupTmpDir = ClientBase.createTmpDir();
+    File backupDir = ClientBase.createTmpDir();
+
+    BackupManager bm = new BackupManager(dataDir, dataDir, dataDir, backupTmpDir, 15,
+        new LocalBackupStorage(backupConfig));
+    bm.initialize();
+
+    bm.getLogBackup().run(1);
+    bm.getSnapBackup().run(1);
+
+    Assert.assertFalse("No files should have been backed up.", backupDir.list() == null);
```

Comment:
It's auto-generated - `test5659468055413506152.junit.dir` for example.

When the test starts the ZK, it would look something like this:

```
2021-01-22 19:17:45,312 [myid:] - INFO  [main:BackupManager@629] - snapDir=/Users/hulee/mp/zk/zookeeper/zookeeper-server/target/surefire/test5659468055413506152.junit.dir
2021-01-22 19:17:45,312 [myid:] - INFO  [main:BackupManager@630] - dataLogDir=/Users/hulee/mp/zk/zookeeper/zookeeper-server/target/surefire/test5659468055413506152.junit.dir
2021-01-22 19:17:45,312 [myid:] - INFO  [main:BackupManager@631] - backupStatusDir=/Users/hulee/mp/zk/zookeeper/zookeeper-server/target/surefire/test4020572069576308599.junit.dir
2021-01-22 19:17:45,312 [myid:] - INFO  [main:BackupManager@632] - tmpDir=/Users/hulee/mp/zk/zookeeper/zookeeper-server/target/surefire/test1336151023536551739.junit.dir
2021-01-22 19:17:45,312 [myid:] - INFO  [main:BackupManager@633] - backupIntervalInMinutes=15
Creating file /Users/hulee/mp/zk/zookeeper/zookeeper-server/target/surefire/test4020572069576308599.junit.dir/zkBackupStatus
```